### DISCUSSION TITLE
Framerate Calculation Method

### DISCUSSION THREAD

**Discussion by bchah** [OvenMediaLabs/OvenMediaEngine#1346] [Ideas] [answered]
Hello! The framerate reported by the OME API seems to be the average framerate, rather than the current framerate. This is mostly evident in WebRTC streams where the frame rate could change suddenly. If we start a stream with 1FPS, then change to 15FPS, 30FPS, the expectation from the API is that it would return the new "framerate", but it seems to return more of an "avgFramerate" where it slowly increases over time as the average changes.

**My feature request is**: please consider renaming the current "framerate" stream statistic to "avgFramerate" and introduce a new "framerate" or "currentFramerate" statistic which only calculates the average number frames received between the last 2 keyframes.

What do you think of this? And is there something about the framerate statistics I am not understanding, or another way to access the "true" current framerate? As always, thank you for your amazing work! 🥇

**Comment by getroot** [comment, marked as answer]
Good idea! it will be added to our backlog
